File: DSA/Priyanshi/Day5_que2.java
class Day5_que2 
{
    static Node head1;
    static Node head2;

    static class Node 
    {
        int data;
        Node next;

        Node(int d) {
            data = d;
            next = null;
        }
    }

    // Function to compare two linked lists
    public static String compareLists(Node head1, Node head2) 
    {
        Node temp1 = head1;
        Node temp2 = head2;

        while (temp1 != null && temp2 != null) 
        {
            if (temp1.data != temp2.data) 
            {
                return "different";
            }
            temp1 = temp1.next;
            temp2 = temp2.next;
        }

        // Check if both lists reached the end
        if (temp1 == null && temp2 == null) 
        {
            return "identical";
        } 
        else 
        {
            return "different";
        }
    }

    //function to create a linked list from an array of values
    public static Node createLinkedList(int[] values) 
    {
        if (values.length == 0) return null;
        Node head = new Node(values[0]);
        Node current = head;
        for (int i = 1; i < values.length; i++) 
        {
            current.next = new Node(values[i]);
            current = current.next;
        }
        return head;
    }

    public static void main(String[] args) 
    {
        int[] values1 = {1, 2, 3};
        head1 = createLinkedList(values1);

        int[] values2 = {1, 2, 3};
        head2 = createLinkedList(values2);

        String result = compareLists(head1, head2);
        System.out.println("The linked lists are " + result);
    }
}

File: DSA/Devansh-Nishchal-Mandloi/day5que2.cpp
#include <iostream>

using namespace std;

class Node {
public:
    int data;
    Node* next;

    Node(int val) : data(val), next(nullptr) {}
};

class LinkedList {
private:
    Node* head;

public:
    LinkedList() : head(nullptr) {}

    void append(int val) {
        Node* newNode = new Node(val);
        if (head == nullptr) {
            head = newNode;
        } else {
            Node* current = head;
            while (current->next != nullptr) {
                current = current->next;
            }
            current->next = newNode;
        }
    }

    Node* getHead() {
        return head;
    }

    void printList() {
        Node* current = head;
        while (current != nullptr) {
            cout << current->data << " -> ";
            current = current->next;
        }
        cout << "null" << endl;
    }
};

string compareLists(Node* head1, Node* head2) {
    Node* current1 = head1;
    Node* current2 = head2;

    while (current1 != nullptr && current2 != nullptr) {
        if (current1->data != current2->data) {
            return "different";
        }
        current1 = current1->next;
        current2 = current2->next;
    }

    if (current1 != nullptr || current2 != nullptr) {
        return "different";
    }

    return "identical";
}

int main() {
    LinkedList list1;
    list1.append(1);
    list1.append(2);
    list1.append(3);

    LinkedList list2;
    list2.append(1);
    list2.append(2);
    list2.append(3);

    cout << "Linked list 1: ";
    list1.printList();

    cout << "Linked list 2: ";
    list2.printList();

    string result = compareLists(list1.getHead(), list2.getHead());
    cout << "Output: " << result << endl;

    return 0;
}
